src/api/product/data/product.mapper.ts

```diff
@@ -0,0 +1,22 @@
+import Product from './product.entity';
+import { IProductRepository } from '../shared/product.types';
+
+export class ProductMapper {
+	constructor() {}
+
+	public toArrayOfDomain(data: IProductRepository[]): Product[] {
+		return data.map(({ _id, title, description, price }) => ({
+			id: _id,
```

Comment:
If I am not mistaken, mongoose provides `id` virtual getter behind the scenes. It should be basically pointing to _id. Maybe I forgot or things changed. But should be something like this https://mongoosejs.com/docs/guide.html#id